docs/_posts/0000-01-03-continuous-delivery-and-deployment.md

```diff
@@ -253,27 +273,196 @@ Delete the branch… done.
+This situation is not uncommon ;-)
 We're going to do some exercises involving branching, git is a very powerful tool, it's possible to get things VERY wrong.
 
 --
 
+<img style="border:0;box-shadow:none;background-color:transparent" src="images/aliens.png">
+
+Note:
+I'm increasingly of the opinion that there is only one real explanation for how git works, and that is that it is alien technology ;-)
+
+--
+
 <img width="500px" src="images/guitar_hero.png">
 
 Note:
 Branches can easily get out of control, they should be few in number and short-lived - you have been warned.
 
 --
 
-<img style="border:0;box-shadow:none;background-color:transparent" src="images/aliens.png">
+### Git guidelines
+
++ Use a small set of commands:
+    + clone
+    + commit
+    + push
+    + pull
++ Use a consistent workflow
++ Branches should be short-lived
++ The number of branches should be small
+
+--
+
+## Merging
+
+<img src="https://wac-cdn.atlassian.com/dam/jcr:09308632-38a3-4637-bba2-af2110629d56/07.svg?cdnVersion=ks">
+
 
 Note:
-I'm increasingly of the opinion that there is only one real explanation for how git works, and that is that it is alien technology ;-)
+What is a merge
+Sometimes it's easy
+And sometimes it's hard
+
+--
+
+<backgroundimage>images/traffic_merge.svg</backgroundimage>
+<backgroundimageopacity>0.25</backgroundimageopacity>
+
++ Git is actually good at merging
++ Some merges are simple
++ Some merges are not
++ Keeping your branches short-lived decreases the risk of complicated merges
+    
+
+--
+
+
+### Different kinds of merge
+
+* Simple
+    + Different files
+    + Same file, different lines
+    + Same file, same lines
+* Complicated
+
+Note:
+Whiteboard what might be simple and complex merges with the group
+
+--
+
+### Diverging branches
+
+<img src="images/diverging_branches.svg">
+
++ The problem gets worse with distance and time
+
+
+Note:
+It's not just the change itself that determines the complexity of the merge, they get worse with distance and time.
+You can end up with separate bits of software.
+Like straddling two diverging canoes.
+
+
+--
+
+### A simple merge
+
+<img src="images/green_merge.png">
+
++ Be a little wary
+
+Note:
+Worth pointing out that just because a merge can be completed automatically, does not guarantee the functionality is
+correct.
+
+--
+
+
+### A complicated merge
+
+<img src="images/merge_hell.png">
+
+Note:
+A salutary tale - whilst working on the slides I failed to follow my own advice and ended up with conflicts.
+Explain the merge view, our changes on the left, base version in the middle, other developer's version on the right.
+
+--
+
+### Conflicts
+
+```
+<<<<<<< HEAD:file.txt
+Conflicts
+=======
+What conflicts?
+>>>>>>> 77976da35a11db4580b80ae27e8d65caf5208086:file.txt
+```
+
++ Must be resolved manually
++ _**BE AFRAID!**_
+
+
+Note:
+What are they
+What do they look like
+What is the effect
+How do you resolve them
```

Comment:
These notes don’t render properly. Remember you need two spaces at the end of Note lines to force a line break.

Reply:
Will fix - didn't know about the space hack